source/common/router/header_formatter.cc

```diff
@@ -245,6 +245,14 @@ StreamInfoHeaderFormatter::StreamInfoHeaderFormatter(absl::string_view field_nam
     field_extractor_ = [](const StreamInfo::StreamInfo& stream_info) -> std::string {
       return std::string(stream_info.downstreamAddressProvider().requestedServerName());
     };
+  } else if (field_name == "VIRTUAL_CLUSTER_NAME") {
+    field_extractor_ = [](const Envoy::StreamInfo::StreamInfo& stream_info) -> std::string {
+      absl::optional<std::string> virtual_cluster_name = stream_info.virtualClusterName();
```

Comment:
can you try replacing the body of this lambda with:
```
return stream_info.virtualClusterName().value_or("");
```
I think that should work, though maybe you might need
```
return stream_info.virtualClusterName().value_or(std::string(""));
```

See https://github.com/abseil/abseil-cpp/blob/9336be04a242237cd41a525bedfcf3be1bb55377/absl/types/optional.h#L503

Reply:
Done, Thank You!